tests/test_helpers.py

```diff
@@ -167,10 +147,9 @@ def format_error(error):
     assert isinstance(output.body, str)
     assert isinstance(output.status_code, int)
     assert json.loads(output.body) == {
-        "data": None,
         "errors": [{"msg": "Some msg", "loc": "1:2", "pth": "some/path"}],
     }
-    assert output.status_code == 200
+    assert output.status_code == 400
```

Comment:
This seems like a subtly nontrivial change... if existing clients expect an HTTP 200 even if the `errors` key is present in the response, won't this potentially break them? Personally, I prefer this behavior, but it seems risky to change outright and I don't see anything that mentions the rationale for doing so.

Reply:
Good spot @wyattanderson ! @KingDarBoja can we revert the change to return a 400 status code if the response contains errors? We should always return with a 200 because there might be partial data returned.

Reply:
The changes were made at these [lines](https://github.com/graphql-python/graphql-server-core/pull/36/files?file-filters%5B%5D=.py&hide-deleted-files=true#diff-0b0d844d3af70da5de18ae457e73d2c2R290-R298) so should be easier to revert.

I do prefer the 400 status code as it reflects there was an error on the request but I didn't took into account the possible side effects for those clients than expect a HTTP 200 response.

Maybe we can keep the new behaviour and put on the readme a **Breaking Change Note** about this as this will be part of `graphql-server-core` v3.